keras_contrib/applications/ror.py

```diff
@@ -0,0 +1,323 @@
+    else:
+        if not K.is_keras_tensor(input_tensor):
+            img_input = Input(tensor=input_tensor, shape=input_shape)
+        else:
+            img_input = input_tensor
+
+    x = __create_pre_residual_of_residual(classes, img_input, include_top,
+                                          depth, width, dropout_rate)
+
+    # Ensure that the model takes into account
+    # any potential predecessors of `input_tensor`.
+    if input_tensor is not None:
+        inputs = get_source_inputs(input_tensor)
+    else:
+        inputs = img_input
+    # Create model.
+    model = Model(inputs, x, name='residual_of_residual')
+
+    # load weights
+    if weights == 'cifar10':
+        if (depth == 40) and (width == 2) and (dropout_rate == 0.0):
+            # Default parameters match. Weights for this model exist:
+
+            if K.image_dim_ordering() == 'th':
+                if include_top:
+                    weights_path = get_file('densenet_40_12_th_dim_ordering_th_kernels.h5',
+                                            TH_WEIGHTS_PATH,
+                                            cache_subdir='models')
+                else:
+                    weights_path = get_file('densenet_40_12_th_dim_ordering_th_kernels_no_top.h5',
+                                            TH_WEIGHTS_PATH_NO_TOP,
+                                            cache_subdir='models')
+
+                model.load_weights(weights_path)
+
+                if K.backend() == 'tensorflow':
+                    warnings.warn('You are using the TensorFlow backend, yet you '
+                                  'are using the Theano '
+                                  'image dimension ordering convention '
+                                  '(`image_dim_ordering="th"`). '
+                                  'For best performance, set '
+                                  '`image_dim_ordering="tf"` in '
+                                  'your Keras config '
+                                  'at ~/.keras/keras.json.')
+                    convert_all_kernels_in_model(model)
+            else:
+                if include_top:
+                    weights_path = get_file('densenet_40_12_tf_dim_ordering_tf_kernels.h5',
+                                            TF_WEIGHTS_PATH,
+                                            cache_subdir='models')
+                else:
+                    weights_path = get_file('densenet_40_12_tf_dim_ordering_tf_kernels_no_top.h5',
+                                            TF_WEIGHTS_PATH_NO_TOP,
+                                            cache_subdir='models')
+
+                model.load_weights(weights_path)
+
+                if K.backend() == 'theano':
+                    convert_all_kernels_in_model(model)
+
+    return model
+
+
+def __initial_conv(input):
+    x = Convolution2D(16, 3, 3, border_mode='same', init='he_normal')(input)
+    return x
+
+
+def __conv1_block(input, k=1, dropout=0.0, initial=False):
```

Comment:
Actually, model doesn't build correctly if I remove it. It is simpler just to keep it.